### PR TITLE
feat(useContextMenu): new function

### PR DESCRIPTION
**Pull request by vaakian**
### Description

Add custom [`contextMenu`](https://developer.mozilla.org/en-US/docs/Web/API/Element/contextmenu_event) to an Element. 

#### this PR depends on 2 changes:
* https://github.com/vueuse/vueuse/pull/2143
* https://github.com/vueuse/vueuse/pull/2161

https://user-images.githubusercontent.com/30516060/187668848-80200739-b3ce-4430-bc95-c930236816f7.mov

### Additional context
I closed the [previous PR](https://github.com/vueuse/vueuse/pull/2136) because the commit history is too messy.
**And of course feel free to correct my English grammar issues in the files~** 🤪

---

### What is the purpose of this pull request? 

- [ ] Bug fix
- [x] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

**Review by songjianet — CHANGES_REQUESTED** [review]
You don't have to close the previous pr. If it is closed, you need to review it again. Can you do some unit testing on this feature and fix the previous review. https://github.com/vueuse/vueuse/pull/2136

**Comment by vaakian** [comment]
> You don't have to close the previous pr. If it is closed, you need to review it again. Can you do some unit testing on this feature and fix the previous review. #2136

A unit testing file is included in the latest commit. but t's quite verbose in some way(:

File: packages/core/useContextMenu/index.md
Comment (vaakian) [reply packages/core/useContextMenu/index.md:78]: Oh, sure! I didn't notice it when pasting that piece of code from demo.

**Comment by vaakian** [comment]
> I think any changes to other composables(e.g. useMouseInElement) should be done in separate PRs

I thought the same, but this PR needs those changes to make the new function work, so I must wait until those separate PRs are merged before opening this PR, otherwise the CI would fail.

Sorry about my terrible git skill, is there a decent work around to do this?

**Comment by wheatjs** [comment]
It would be best to open up the other PRs and just make a note of it in this one that it depends on the other PRs. Once those are merged you can then bring this on up to date with main and then the tests will pass

File: packages/core/useContextMenu/index.ts
Comment (vaakian) [inline packages/core/useContextMenu/index.ts:211]: 💡 Got an idea about `useEventListener`
```ts
// multi event
useEventListener(['scroll', 'click', 'resize', 'blur'], handler)

// multi handler
useEventListener('scroll' [handler1, handler2, handler3])

// multi event & multi handler
useEventListener(['scroll', 'click', 'resize', 'blur'], [handler1, handler2, handler3])
```

Comment (vaakian) [reply packages/core/useContextMenu/index.ts:211]: seems not to be the right place, I will place this idea somewhere else later on.😂

**Comment by vaakian** [comment]
Oh! how did I just accidentially closed this PR, sorry for the disturbance.

**Pull request closed without merging**
